src/SFA.DAS.EmploymentCheck.Functions/Application/Services/EmploymentCheck/EmploymentCheckServiceBase.cs

```diff
@@ -567,92 +473,152 @@ await using (sqlConnection)
+                            // No paye schems found for this apprentice so we're not able to do an employment and will need to skip this apprentice
+                            logger.LogInformation($"{thisMethodName}: {ErrorMessagePrefix} No PAYE schemes found for apprentice Uln: [{employmentCheck.Uln}] accountId [{employmentCheck.AccountId}].");
                             continue;
                         }
 
                         // Create the individual message combinations for each paye scheme
                         foreach (var payeScheme in employerPayeSchemes.PayeSchemes)
                         {
-                            apprenticeEmploymentCheckMessage.EmploymentCheckId = apprentice.Id;
-                            apprenticeEmploymentCheckMessage.Uln = apprentice.ULN;
-                            apprenticeEmploymentCheckMessage.NationalInsuranceNumber = nationalInsuranceNumber;
-                            apprenticeEmploymentCheckMessage.StartDateTime = apprentice.MinDate;
-                            apprenticeEmploymentCheckMessage.EndDateTime = apprentice.MinDate;
-                            apprenticeEmploymentCheckMessage.PayeScheme = payeScheme;
-
-                            apprenticeEmploymentCheckMessages.Add(apprenticeEmploymentCheckMessage);
+                            if(employmentCheck.Id > 0 &&
+                                employmentCheck.Uln > 0 &&
+                                !string.IsNullOrEmpty(nationalInsuranceNumber) &&
+                                employmentCheck.MinDate > DateTime.MinValue &&
+                                employmentCheck.MaxDate > DateTime.MinValue &&
+                                !string.IsNullOrEmpty(payeScheme))
+                            {
+                                var employmentCheckMessage = new EmploymentCheckMessage();
+
+                                employmentCheckMessage.EmploymentCheckId = employmentCheck.Id;
+                                employmentCheckMessage.Uln = employmentCheck.Uln;
+                                employmentCheckMessage.NationalInsuranceNumber = nationalInsuranceNumber;
+                                employmentCheckMessage.MinDateTime = employmentCheck.MinDate;
+                                employmentCheckMessage.MaxDateTime = employmentCheck.MaxDate;
+                                employmentCheckMessage.PayeScheme = payeScheme;
+
+                                employmentCheckMessages.Add(employmentCheckMessage);
+                            }
+                            else
+                            {
+                                // There was invalid data preventing a message being created
+                                logger.LogInformation($"{thisMethodName}: {ErrorMessagePrefix} One of the following input values for creating an EmploymentCheckMessage was invalid: \n" +
```

Comment:
This logging can be made better to log specifically which variable caused the issue. Not the most pressing but something we'll want to do at some point

Reply:
The code has moved on considerably since this PR was raised and the file (EmploymentCheckServiceBase) no longer exists as it has been merged into the EmploymentCheckService file. In the EmploymentCheckService I am implemented a validator class to simply the validation checking and log the relevant invalid fields in a class as shown below:

           IList<EmploymentCheckCacheRequest> employmentCheckCacheRequests = new List<EmploymentCheckCacheRequest>();
            EmploymentCheckModelValidator employmentCheckModelValidator = new EmploymentCheckModelValidator();
            try
            {
                if(employmentCheckModels != null && employmentCheckModels.Any())
                {
                    foreach(var employmentCheckModel in employmentCheckModels)
                    {
                        var validationResult = employmentCheckModelValidator.Validate(employmentCheckModel);

                        if(validationResult != null &&
                            validationResult.IsValid)
                        {
                            var employmentCheckCacheRequest = new EmploymentCheckCacheRequest();

                            employmentCheckCacheRequest.EmploymentCheckId = employmentCheckModel.Id;
                            employmentCheckCacheRequest.CorrelationId = employmentCheckModel.CorrelationId;
                            employmentCheckCacheRequest.RequestTypeId = (byte)RequestTypeEnum.EmploymentCheck;
                            employmentCheckCacheRequest.CreatedOn = DateTime.Now;
                            employmentCheckCacheRequest.LastUpdated = employmentCheckCacheRequest.CreatedOn;

                            employmentCheckCacheRequests.Add(employmentCheckCacheRequest);
                        }
                        else
                        {
                            foreach(var error in validationResult.Errors)
                            {
                                _logger.LogError($"{thisMethodName}: {ErrorMessagePrefix} EmploymentCheckMode {error.ErrorMessage}");
                            }
                        }
                    }
                }
            }

So can you mark this as resolved please so I can raise a new PR with the changed code?